servicetalk-client-api/src/main/java/io/servicetalk/client/api/AutoRetryStrategyProvider.java

```diff
@@ -32,15 +33,18 @@
     /**
      * An {@link AutoRetryStrategyProvider} that disables automatic retries;
      */
-    AutoRetryStrategyProvider DISABLE_AUTO_RETRIES = __ -> (___, cause) -> failed(cause);
+    AutoRetryStrategyProvider DISABLE_AUTO_RETRIES = (lbEventStream, sdErrorStream) -> (___, cause) -> failed(cause);
 
     /**
-     * Create a new {@link AutoRetryStrategy} instance using the passed {@link LoadBalancer}.
+     * Creates a new {@link AutoRetryStrategy} instance.
      *
-     * @param loadBalancer {@link LoadBalancer} to use.
+     * @param lbEventStream a stream of events from {@link LoadBalancer#eventStream() LoadBalancer}
+     * @param sdStatus a {@link Completable} that reports current {@link ServiceDiscoverer} status. It fails when
+     * {@link ServiceDiscoverer#discover(Object) discovery publisher} emits an error or completes when the publisher
```

Comment:
As we do not expect SD streams to complete, can we not mention the completion contract here?

If we complete the `Completable` when SD completes, any request waiting may prematurely be retried before LB being ready.